File: app/src/main/java/com/example/lpiem/hearthstonecollectorapp/Activities/NavigationActivity.java
package com.example.lpiem.hearthstonecollectorapp.Activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;
import android.view.View;
import android.support.design.widget.NavigationView;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.lpiem.hearthstonecollectorapp.Fragments.CardsListFragment;
import com.example.lpiem.hearthstonecollectorapp.R;
import com.facebook.login.LoginManager;
import com.squareup.picasso.Picasso;

import org.json.JSONObject;

public class NavigationActivity extends AppCompatActivity
        implements NavigationView.OnNavigationItemSelectedListener {
    JSONObject response, profile_pic_data, profile_pic_url;

    private DrawerLayout drawerLayout;
    private NavigationView navigationView;

    private Fragment cardsListFragment;
    private Fragment decksFragment;
    private Fragment tradeFragment;
    private Fragment quizzFragment;
    private Fragment shopFragment;

    private static final int FRAGMENT_CARDSLIST = 0;
    private static final int FRAGMENT_DECKS = 1;
    private static final int FRAGMENT_TRADE = 2;
    private static final int FRAGMENT_QUIZZ = 3;
    private static final int FRAGMENT_SHOP = 4;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_navigation);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        this.drawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                this, drawerLayout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawerLayout.addDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);

        Intent intent = getIntent();
        String jsondata = intent.getStringExtra("userProfile");
        Log.w("Jsondata", jsondata);
        View headerView = navigationView.getHeaderView(0);
        TextView user_name = (TextView) headerView.findViewById(R.id.nameUser);
        ImageView user_picture = (ImageView) headerView.findViewById(R.id.imgUser);

        try {
            response = new JSONObject(jsondata);
            Log.e("[NavigationActivity]", response.get("name").toString());
            user_name.setText(response.get("name").toString());

            if(!response.isNull("pictureUrlGoogle") && response.has("pictureUrlGoogle")){
                System.out.println("[NavigationActivity] url picture google ok");
                profile_pic_url = new JSONObject();
                profile_pic_url.put("url", response.getString("pictureUrlGoogle"));
            } else {
                profile_pic_data = new JSONObject(response.get("picture").toString());
                profile_pic_url = new JSONObject(profile_pic_data.getString("data"));
            }

            Picasso.with(this).load(profile_pic_url.getString("url"))
                    .into(user_picture);

        } catch(Exception e){
            e.printStackTrace();
        }
    }

    @Override
    public void onBackPressed() {
        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START);
        } else {
            super.onBackPressed();
        }
    }

    @SuppressWarnings("StatementWithEmptyBody")
    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        int id = item.getItemId();

        if (id == R.id.nav_cards) {
            this.showFragment(FRAGMENT_CARDSLIST);
        } else if (id == R.id.nav_decks) {
            //showFragment(new DecksFragment());
        } else if (id == R.id.nav_trade) {
           // showFragment(new TradeFragment());
        } else if (id == R.id.nav_quizz) {
            //showFragment(new QuizzFragment());
        } else if (id == R.id.nav_shop) {
            //showFragment(new ShopFragment());
        } else if (id == R.id.nav_deconnexion) {
            Intent intent = new Intent(NavigationActivity.this, ConnexionActivity.class);
            intent.putExtra("deconnexion", true);
            startActivity(intent);
        }

        this.drawerLayout.closeDrawer(GravityCompat.START);
        return true;
    }

    public void showFragment(int fragmentIdentifier) {
        switch (fragmentIdentifier){
            case FRAGMENT_CARDSLIST :
                this.showCardsListFragment();
                break;
            default:
                break;
        }
    }

    private void showCardsListFragment(){
        if (this.cardsListFragment == null) this.cardsListFragment = CardsListFragment.newInstance();
        this.startTransactionFragment(this.cardsListFragment);
    }

    private void startTransactionFragment(Fragment fragment){
        if (!fragment.isVisible()){
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.drawer_layout, fragment).commit();
        }
    }


}

File: app/src/main/java/com/example/lpiem/hearthstonecollectorapp/Activities/NavigationActivity.kt
package com.example.lpiem.hearthstonecollectorapp.Activities

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import com.google.android.material.navigation.NavigationView
import androidx.core.view.GravityCompat
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import android.view.MenuItem
import com.bumptech.glide.Glide
import android.widget.FrameLayout
import com.example.lpiem.hearthstonecollectorapp.Fragments.CardsListFragment
import com.example.lpiem.hearthstonecollectorapp.Manager.HsUserManager
import com.example.lpiem.hearthstonecollectorapp.Fragments.DecksListFragment
import com.example.lpiem.hearthstonecollectorapp.R
import kotlinx.android.synthetic.main.activity_navigation.*
import kotlinx.android.synthetic.main.nav_header_navigation.view.*
import org.json.JSONObject

class NavigationActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener {
    internal lateinit var response: JSONObject
    internal lateinit var profile_pic_data:JSONObject
    internal lateinit var profile_pic_url:JSONObject

    private var hsUserManager = HsUserManager

    private var drawerLayout: androidx.drawerlayout.widget.DrawerLayout? = null
    private var content: FrameLayout? = null

    private val decksFragment: androidx.fragment.app.Fragment? = null
    private val tradeFragment: androidx.fragment.app.Fragment? = null
    private val quizzFragment: androidx.fragment.app.Fragment? = null
    private val shopFragment: androidx.fragment.app.Fragment? = null

    private val FRAGMENT_CARDSLIST = 0
    private val FRAGMENT_DECKS = 1
    private val FRAGMENT_TRADE = 2
    private val FRAGMENT_QUIZZ = 3
    private val FRAGMENT_SHOP = 4

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_navigation)

        content = findViewById(R.id.content_navigation) as FrameLayout

//        val toggle = ActionBarDrawerToggle(
//                this, drawer_layout, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)

        Log.d("InNav-User", hsUserManager.loggedUser.toString())
        Log.d("InNav-UserSocialInfos", hsUserManager.userSocialInfos.toString())

        val toggle = ActionBarDrawerToggle(
                this, drawer_layout, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()

        nav_view.setNavigationItemSelectedListener(this)

        val headerView = nav_view.getHeaderView(0)

        try {
            Log.d("[NavigationActivity]", hsUserManager.loggedUser.pseudo)
            headerView.nameUser.text = hsUserManager.loggedUser.pseudo

            if (hsUserManager.userSocialInfos.isNull("email")) {
                Log.d("NavigationActivity", "No social email, add default avatar")
                setHeaderAvatar(hsUserManager.defautThumbnail)
            } else {
                if (hsUserManager.userSocialInfos.get("email") != null) {
                    if (hsUserManager.userSocialInfos.get("type") == "f") { //On est connecté via Facebook
                        if (hsUserManager.userSocialInfos.get("picture") != null) { //Récupérer l'image de FB
                            var json = hsUserManager.userSocialInfos.get("picture") as JSONObject
                            var data = json.get("data") as JSONObject
                            var url = data.get("url") as String
                            Log.d("Facebook Image URL", url)
                            setHeaderAvatar(url)
                        } else {
                            setHeaderAvatar(hsUserManager.defautThumbnail)
                        }
                    } else if (hsUserManager.userSocialInfos.get("type") == "g") {
                        if (hsUserManager.userSocialInfos.get("picture") != null) { //Récupérer l'image de google
                            var url = hsUserManager.userSocialInfos.get("picture") as Uri
                            Log.d("Google Image URL", url.toString())
                            setHeaderAvatar(url.toString())
                            //Picasso.with(this).load(url).into(imgUser)
                        } else {
                            setHeaderAvatar(hsUserManager.defautThumbnail)
                        }
                    }
                }
            }

            nav_view.getHeaderView(0).moneyUser.text = hsUserManager.loggedUser.coins.toString()
        } catch (e: Exception) {
            e.printStackTrace()
        }

        // Fragment par défaut : liste des cartes
        val fragment = CardsListFragment.newInstance()
        replaceFragment(fragment)
        nav_view.getMenu().getItem(0).setChecked(true)
    }

    fun setHeaderAvatar(image: String) {
        Glide.with(this).load(image).into(nav_view.getHeaderView(0).imgUser)
    }

    override fun onBackPressed() {
        if (drawer_layout.isDrawerOpen(GravityCompat.START)) {
            drawer_layout.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        return super.onOptionsItemSelected(item)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.nav_cards -> {
                val fragment = CardsListFragment() //.newInstance()
                replaceFragment(fragment)
            }
            R.id.nav_decks -> {
                val fragment = DecksListFragment() //.newInstance()
                replaceFragment(fragment)
            }
            R.id.nav_trade -> {
                // showFragment(new TradeFragment());
            }
            R.id.nav_quizz -> {
                //showFragment(new QuizzFragment());
            }
            R.id.nav_shop -> {
                //showFragment(new ShopFragment());
            }
            R.id.nav_deconnexion -> {
                val intent = ConnexionActivity.newIntent(this, true)
                startActivity(intent)
            }
        }

        drawer_layout.closeDrawer(GravityCompat.START)
        return true
    }

    private fun replaceFragment(fragment: androidx.fragment.app.Fragment) {
        supportFragmentManager
                .beginTransaction()
                .replace(R.id.content_navigation, fragment, fragment.javaClass.getSimpleName())
                .addToBackStack(fragment.javaClass.getSimpleName())
                .commit()
    }


}
